config/routes.rb

```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  get 'users', to: 'users#index'
+  get 'users/:id', to: 'users#show', as: 'user'
+  get 'users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get 'users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
 end
```

Comment:
- [Optional] Dear @Caren-Koroeny, Your routes are good but can be improved further as a bit less verbose.
```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```